src/main/java/com/doctusoft/math/ExponentialDelays.java

```diff
@@ -0,0 +1,438 @@
+package com.doctusoft.math;
+
+import com.doctusoft.annotation.Beta;
+import com.doctusoft.java.Failsafe;
+
+import java.util.*;
+import java.util.function.*;
+import java.util.stream.*;
+
+import static java.util.Objects.*;
+
+/**
+ * This utility helps create streams of exponentially increasing int or long values complying with various criteria.
+ * Assembling all those criteria is done in a builder style defined by {@link OfInt} and {@link OfLong} internal
+ * interfaces. Invoking their {@code build()} method returns a new {@link IntStream} or {@link LongStream} instance
+ * with the desired attributes.
+ */
+@Beta
+public interface ExponentialDelays {
```

Comment:
I believe the first argument is only valid upon overloading the system thus braking it. I can think of use-cases where this is not the case or cannot happen realisticly (e.g. GAE deploy status check), but hitting the 0,1% error rate is rather likely with one request (e.g. reading from a GCS bucket). In such cases unrandomized delays should not make much of a difference.

However I see an opportunity to supporting randomized delays as well with another class called ExponentialRandomizedDelays.

The second suggestion is something I was thinking of originally as well but had been left out from the first implementation, mostly because it was not yet used in existing code. I also could not come up with a suitable name, which certainly is not an issue anymore since limitOverallExecution is an excellent suggestion. I will create 2 separate issues for this.